infra/modules/s3/main.tf

```diff
@@ -67,4 +67,65 @@ resource "aws_s3_bucket_acl" "email_forwarding_bucket" {
   ]
 }
 EOF
-}
\ No newline at end of file
+}
+
+resource "aws_lambda_function" "pdf_processing_lambda" {
+  filename      = "${path.module}/pdf-processing-lambda.zip"
+  function_name = "${var.environment}-pdf-processing-lambda"
+  role          = aws_iam_role.pdf_processing_lambda_role.arn
+  handler       = "lambda_function.lambda_handler"
+  runtime       = "nodejs18.x"
+  source_code_hash = filebase64sha256("${path.module}/pdf-processing-lambda.zip")
+
+  environment {
+    variables = {
+      AWS_REGION             = data.aws_region.default.name
```

Comment:
When I run terraform plan it complains about this undeclared resource:

'A data resource "aws_region" "default" has not been declared in module.s3.'

Reply:
The correct way to do it is in the fixed inboxes branch...

@shaunajisc I think you need to take this PR back and make sure it is working and aligns with the OCT-689 fixes

Reply:
Will do 👍🏻